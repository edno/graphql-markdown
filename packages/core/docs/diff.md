# diff

## Functions

### hasChanges()

```ts
function hasChanges(
   schema, 
   tmpDir, 
   diffMethod, 
diffModule): Promise<boolean>
```

#### Parameters

• **schema**: `GraphQLSchema`

• **tmpDir**: `string`

• **diffMethod**: `Maybe`\<`DiffMethodName`\>

• **diffModule**: `Maybe`\<`string`\> = `"@graphql-markdown/diff"`

#### Returns

`Promise`\<`boolean`\>

#### Defined in

[diff.ts:10](https://github.com/graphql-markdown/graphql-markdown/blob/main/packages/core/src/diff.ts#L10)
