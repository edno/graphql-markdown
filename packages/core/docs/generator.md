# generator

## Functions

### generateDocFromSchema()

```ts
function generateDocFromSchema(__namedParameters): Promise<void>
```

#### Parameters

â€¢ **\_\_namedParameters**: `GeneratorOptions`

#### Returns

`Promise`\<`void`\>

#### Source

[generator.ts:28](https://github.com/graphql-markdown/graphql-markdown/blob/main/packages/core/src/generator.ts#L28)
