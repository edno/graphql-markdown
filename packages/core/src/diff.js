const logger = require("@graphql-markdown/utils").logger.getInstance();

const hasChanges = async (
  schema,
  tmpDir,
  diffMethod,
  diffModule = "@graphql-markdown/diff",
) => {
  if (
    typeof diffMethod === "undefined" ||
    diffMethod == null ||
    typeof diffModule === "undefined" ||
    diffModule == null
  ) {
    return true;
  }

  try {
    const { checkSchemaChanges } = require(diffModule);
    return await checkSchemaChanges(schema, tmpDir, diffMethod);
  } catch (error) {
    logger.warn(
      `Cannot find module '${diffModule}' from @graphql-markdown/core!`,
    );
  }

  return true;
};

module.exports = { hasChanges };
