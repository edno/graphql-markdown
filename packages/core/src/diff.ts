import type {
  DiffMethodName,
  FunctionCheckSchemaChanges,
  GraphQLSchema,
  Maybe,
} from "@graphql-markdown/types";

import { Logger } from "@graphql-markdown/utils";

const logger = Logger.getInstance();

export const hasChanges = async (
  schema: GraphQLSchema,
  tmpDir: string,
  diffMethod: DiffMethodName,
  diffModule: Maybe<string> = "@graphql-markdown/diff",
): Promise<boolean> => {
  if (
    typeof diffMethod === "undefined" ||
    diffMethod == null ||
    typeof diffModule === "undefined" ||
    diffModule == null
  ) {
    return true;
  }

  try {
    const {
      checkSchemaChanges,
    }: { checkSchemaChanges: FunctionCheckSchemaChanges } = await import(
      diffModule
    );
    return await checkSchemaChanges(schema, tmpDir, diffMethod);
  } catch (error) {
    logger.warn(
      `Cannot find module '${diffModule}' from @graphql-markdown/core!`,
    );
    return true;
  }
};
