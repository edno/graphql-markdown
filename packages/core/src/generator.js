const {
  graphql: { getSchemaMap, loadSchema, getDocumentLoaders },
} = require("@graphql-markdown/utils");
const { getGroups } = require("./group-info");
const Renderer = require("./renderer");

const NS_PER_SEC = 1e9;

const hasChanges = async (
  schema,
  tmpDir,
  diffMethod,
  diffModule = "@graphql-markdown/diff",
) => {
  if (
    typeof diffMethod === "undefined" ||
    diffMethod == null ||
    typeof diffModule === "undefined" ||
    diffModule == null
  ) {
    return true;
  }

  try {
    const { checkSchemaChanges } = require(diffModule);
    return await checkSchemaChanges(schema, tmpDir, diffMethod);
  } catch (error) {
    console.warn(
      `Cannot find module '${diffModule}' from @graphql-markdown/core!`,
    );
  }

  return true;
};

const getPrinter = (
  schema,
  baseURL,
  linkRoot,
  groups,
  printTypeOptions,
  printerModule,
  skipDocDirective,
) => {
  let Printer = undefined;

  if (typeof printerModule !== "string") {
    throw new Error(
      'Invalid printer module name in "printTypeOptions" settings.',
    );
  }

  try {
    Printer = require(printerModule);
  } catch (error) {
    throw new Error(
      `Cannot find module '${printerModule}' for @graphql-markdown/core in "printTypeOptions" settings.`,
    );
  }

  Printer.init(schema, baseURL, linkRoot, {
    groups,
    printTypeOptions,
    skipDocDirective,
  });

  return Printer;
};

const generateDocFromSchema = async ({
  baseURL,
  schemaLocation,
  outputDir,
  linkRoot,
  homepageLocation,
  diffMethod,
  tmpDir,
  loaders: loadersList,
  groupByDirective,
  prettify,
  docOptions,
  printTypeOptions,
  printer: printerModule,
  skipDocDirective,
}) => {
  const start = process.hrtime.bigint();

  const loaders = getDocumentLoaders(loadersList);
  const schema = await loadSchema(schemaLocation, loaders);

  const changed = await hasChanges(schema, tmpDir, diffMethod);
  if (!changed) {
    console.info(`No changes detected in schema "${schemaLocation}".`);
  }

  const rootTypes = getSchemaMap(schema);
  const groups = new getGroups(rootTypes, groupByDirective);
  const printer = getPrinter(
    schema,
    baseURL,
    linkRoot,
    groups,
    printTypeOptions,
    printerModule,
    skipDocDirective,
  );
  const renderer = new Renderer(
    printer,
    outputDir,
    baseURL,
    groups,
    prettify,
    docOptions,
  );

  const pages = await Promise.all(
    Object.keys(rootTypes).map((typeName) =>
      renderer.renderRootTypes(typeName, rootTypes[typeName]),
    ),
  );

  await renderer.renderHomepage(homepageLocation);

  const sidebarPath = await renderer.renderSidebar();

  const duration = (
    Number(process.hrtime.bigint() - start) / NS_PER_SEC
  ).toFixed(3);

  console.info(
    `Documentation successfully generated in "${outputDir}" with base URL "${baseURL}".`,
  );
  console.log(
    `${
      pages.flat().length
    } pages generated in ${duration}s from schema "${schemaLocation}".`,
  );
  console.info(
    `Remember to update your Docusaurus site's sidebars with "${sidebarPath}".`,
  );
};

module.exports = { getPrinter, hasChanges, generateDocFromSchema };
