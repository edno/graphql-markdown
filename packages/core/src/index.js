/* istanbul ignore file */

module.exports = {
  generateDocFromSchema: require("./lib/generator").generateDocFromSchema,
  groupInfo: require("./lib/group-info"),
  config: require("./config"),
};
