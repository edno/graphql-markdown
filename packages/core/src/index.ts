/* istanbul ignore file */

export { generateDocFromSchema } from "./generator";
export * from "./config";
