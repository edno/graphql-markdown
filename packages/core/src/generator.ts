import type {
  DiffMethodName,
  GeneratorOptions,
  SchemaEntity,
} from "@graphql-markdown/types";

import { Logger, log } from "@graphql-markdown/logger";

import {
  getCustomDirectives,
  getDocumentLoaders,
  getGroups,
  getSchemaMap,
  loadSchema,
} from "@graphql-markdown/graphql";

import { Renderer } from "./renderer";
import { hasChanges } from "./diff";
import { getPrinter } from "./printer";
import { DiffMethod } from "./config";

const NS_PER_SEC = 1e9 as const;
const SEC_DECIMALS = 3 as const;

export const generateDocFromSchema = async ({
  baseURL,
  customDirective,
  diffMethod,
  docOptions,
  groupByDirective,
  homepageLocation,
  linkRoot,
  loaders: loadersList,
  loggerModule,
  outputDir,
  prettify,
  printer: printerModule,
  printTypeOptions,
  schemaLocation,
  skipDocDirective,
  tmpDir,
}: GeneratorOptions): Promise<void> => {
  const start = process.hrtime.bigint();

  Logger(loggerModule);

  const loaders = await getDocumentLoaders(loadersList);

  if (!loaders) {
    log(
      `An error occurred while loading GraphQL loader.\nCheck your dependencies and configuration.`,
      "error",
    );
    return;
  }

  const schema = await loadSchema(schemaLocation as string, loaders);

  if (diffMethod !== DiffMethod.NONE) {
    const changed = await hasChanges(
      schema,
      tmpDir,
      diffMethod as DiffMethodName,
    );
    if (!changed) {
      log(`No changes detected in schema "${String(schemaLocation)}".`);
    }
  }

  const rootTypes = getSchemaMap(schema);
  const customDirectives = getCustomDirectives(rootTypes, customDirective);
  const groups = getGroups(rootTypes, groupByDirective);
  const printer = await getPrinter(
    // module mandatory
    printerModule,

    // config mandatory
    {
      baseURL,
      linkRoot,
      schema,
    },

    // options
    {
      customDirectives,
      groups,
      printTypeOptions,
      skipDocDirective,
    },
  );
  const renderer = new Renderer(printer, outputDir, baseURL, groups, prettify, {
    ...docOptions,
    deprecated: printTypeOptions.deprecated,
  });

  const pages = await Promise.all(
    Object.keys(rootTypes).map(async (typeName) => {
      return renderer.renderRootTypes(
        typeName as SchemaEntity,
        rootTypes[typeName as SchemaEntity],
      );
    }),
  );

  await renderer.renderHomepage(homepageLocation);

  const sidebarPath = await renderer.renderSidebar();

  const duration = (
    Number(process.hrtime.bigint() - start) / NS_PER_SEC
  ).toFixed(SEC_DECIMALS);

  log(
    `Documentation successfully generated in "${outputDir}" with base URL "${baseURL}".`,
    "success",
  );
  log(
    `${
      pages.flat().length
    } pages generated in ${duration}s from schema "${String(schemaLocation)}".`,
  );
  log(
    `Remember to update your Docusaurus site's sidebars with "${sidebarPath}".`,
  );
};
