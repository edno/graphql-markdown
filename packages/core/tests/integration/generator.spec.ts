import { join } from "node:path";

import { vol } from "memfs";
jest.mock("fs");
jest.mock("node:fs/promises");

import type {
  ClassName,
  GeneratorOptions,
  MDXString,
} from "@graphql-markdown/types";

jest.mock("@graphql-markdown/printer-legacy");
import { Printer } from "@graphql-markdown/printer-legacy";

jest.mock("@graphql-markdown/diff");
import * as diff from "@graphql-markdown/diff";

import { generateDocFromSchema } from "../../src/generator";
import {
  DEFAULT_OPTIONS,
  DeprecatedOption,
  DiffMethod,
} from "../../src/config";

describe("renderer", () => {
  beforeEach(() => {
    // silent console
    jest.spyOn(global.console, "info").mockImplementation(() => {});
    jest.spyOn(global.console, "error").mockImplementation(() => {});

    jest
      .spyOn(Printer, "printType")
      .mockImplementation((value) => value as MDXString);

    vol.fromJSON({
      "/output": null,
      "/temp_test": null,
      "/assets/generated.md": "Dummy homepage for tweet.graphql",
    });
  });

  afterEach(() => {
    vol.reset();
    jest.restoreAllMocks();
  });

  describe("generateDocFromSchema()", () => {
    test("generates Markdown document structure from GraphQL schema", async () => {
      expect.assertions(1);

      const config: GeneratorOptions = {
        baseURL: "graphql",
        schemaLocation: join(__dirname, "../__data__/tweet.graphql"),
        outputDir: "/output",
        linkRoot: "docs",
        homepageLocation: "/assets/generated.md",
        diffMethod: DiffMethod.NONE,
        tmpDir: "/temp",
        loaders: {
          ["GraphQLFileLoader" as ClassName]:
            "@graphql-tools/graphql-file-loader",
        },
        printer: "@graphql-markdown/printer-legacy",
        printTypeOptions: {
          ...DEFAULT_OPTIONS.printTypeOptions,
          deprecated: DeprecatedOption.DEFAULT,
        },
      } as GeneratorOptions;

      await generateDocFromSchema(config);

      expect(vol.toJSON(config.outputDir, undefined, true)).toMatchSnapshot();
    });

    test('outputs "no schema changed" message when called twice', async () => {
      expect.assertions(1);

      const logSpy = jest.spyOn(console, "info");

      const config: GeneratorOptions = {
        baseURL: "graphql",
        schemaLocation: join(__dirname, "../__data__/tweet.graphql"),
        outputDir: "/output",
        linkRoot: "docs",
        homepageLocation: "/assets/generated.md",
        diffMethod: "SCHEMA-HASH",
        tmpDir: "/temp",
        loaders: {
          ["GraphQLFileLoader" as ClassName]:
            "@graphql-tools/graphql-file-loader",
        },
        printer: "@graphql-markdown/printer-legacy",
        printTypeOptions: {
          ...DEFAULT_OPTIONS.printTypeOptions,
          deprecated: DeprecatedOption.DEFAULT,
        },
      } as GeneratorOptions;

      jest.spyOn(diff, "checkSchemaChanges").mockResolvedValue(false);
      await generateDocFromSchema(config);

      expect(logSpy).toHaveBeenCalledWith(
        `No changes detected in schema "${config.schemaLocation}".`,
      );
    });

    test("Markdown document structure from GraphQL schema is correct when using grouping", async () => {
      expect.assertions(2);

      const config: GeneratorOptions = {
        baseURL: "graphql",
        schemaLocation: join(
          __dirname,
          "../__data__/schema_with_grouping.graphql",
        ),
        outputDir: "/output",
        linkRoot: "docs",
        homepageLocation: "/assets/generated.md",
        diffMethod: DiffMethod.NONE,
        tmpDir: "/temp",
        loaders: {
          ["GraphQLFileLoader" as ClassName]:
            "@graphql-tools/graphql-file-loader",
        },
        groupByDirective: {
          directive: "doc",
          field: "category",
          fallback: "misc",
        },
        printer: "@graphql-markdown/printer-legacy",
        printTypeOptions: {
          ...DEFAULT_OPTIONS.printTypeOptions,
          deprecated: DeprecatedOption.DEFAULT,
        },
      } as GeneratorOptions;

      await generateDocFromSchema(config);

      expect(vol.toJSON(config.outputDir, undefined, true)).toMatchSnapshot();
      expect(vol.toJSON(config.tmpDir, undefined, true)).toMatchSnapshot();
    });
  });
});
