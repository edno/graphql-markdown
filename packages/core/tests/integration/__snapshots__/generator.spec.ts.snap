// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer generateDocFromSchema() Markdown document structure from GraphQL schema is correct when using api hierarchy 1`] = `
{
  "generated.md": "Dummy homepage for tweet.graphql",
  "operations/_category_.yml": "label: Operations
position: 1
className: graphql-markdown-api-section
link: null
collapsible: false
collapsed: false
",
  "operations/directives/_category_.yml": "label: Directives
position: 1
link: null
collapsible: true
collapsed: true
",
  "operations/directives/include.mdx": "include",
  "operations/directives/skip.mdx": "skip",
  "operations/mutations/_category_.yml": "label: Mutations
position: 1
link: null
collapsible: true
collapsed: true
",
  "operations/mutations/add-course.mdx": "add-course",
  "operations/mutations/drop-course.mdx": "drop-course",
  "operations/mutations/update-gpa.mdx": "update-gpa",
  "operations/mutations/update-grade.mdx": "update-grade",
  "operations/queries/_category_.yml": "label: Queries
position: 1
link: null
collapsible: true
collapsed: true
",
  "operations/queries/all-courses.mdx": "all-courses",
  "operations/queries/course.mdx": "course",
  "operations/queries/gpa.mdx": "gpa",
  "operations/queries/math-courses.mdx": "math-courses",
  "operations/queries/science-courses.mdx": "science-courses",
  "operations/queries/search-role.mdx": "search-role",
  "operations/queries/un-weighted-gpa.mdx": "un-weighted-gpa",
  "operations/queries/weighted-gpa.mdx": "weighted-gpa",
  "types/_category_.yml": "label: Types
position: 1
className: graphql-markdown-api-section
link: null
collapsible: false
collapsed: false
",
  "types/directives/_category_.yml": "label: Directives
position: 1
link: null
collapsible: true
collapsed: true
",
  "types/directives/deprecated.mdx": "deprecated",
  "types/directives/doc.mdx": "doc",
  "types/directives/no-doc.mdx": "no-doc",
  "types/directives/one-of.mdx": "one-of",
  "types/directives/public.mdx": "public",
  "types/directives/specified-by.mdx": "specified-by",
  "types/enums/_category_.yml": "label: Enums
position: 1
link: null
collapsible: true
collapsed: true
",
  "types/enums/roles.mdx": "roles",
  "types/interfaces/_category_.yml": "label: Interfaces
position: 1
link: null
collapsible: true
collapsed: true
",
  "types/interfaces/record.mdx": "record",
  "types/objects/_category_.yml": "label: Objects
position: 1
link: null
collapsible: true
collapsed: true
",
  "types/objects/course.mdx": "course",
  "types/objects/department-information.mdx": "department-information",
  "types/objects/semester.mdx": "semester",
  "types/objects/unused-type.mdx": "unused-type",
  "types/scalars/_category_.yml": "label: Scalars
position: 1
link: null
collapsible: true
collapsed: true
",
  "types/scalars/boolean.mdx": "boolean",
  "types/scalars/date.mdx": "date",
  "types/scalars/email-address.mdx": "email-address",
  "types/scalars/id.mdx": "id",
  "types/scalars/int.mdx": "int",
  "types/scalars/phone-number.mdx": "phone-number",
  "types/scalars/string.mdx": "string",
  "types/scalars/time.mdx": "time",
  "types/scalars/url.mdx": "url",
}
`;

exports[`renderer generateDocFromSchema() Markdown document structure from GraphQL schema is correct when using api hierarchy 2`] = `{}`;

exports[`renderer generateDocFromSchema() Markdown document structure from GraphQL schema is correct when using flat hierarchy 1`] = `
{
  "add-course.mdx": "add-course",
  "all-courses.mdx": "all-courses",
  "boolean.mdx": "boolean",
  "course.mdx": "course",
  "date.mdx": "date",
  "department-information.mdx": "department-information",
  "deprecated.mdx": "deprecated",
  "doc.mdx": "doc",
  "drop-course.mdx": "drop-course",
  "email-address.mdx": "email-address",
  "generated.md": "Dummy homepage for tweet.graphql",
  "gpa.mdx": "gpa",
  "id.mdx": "id",
  "include.mdx": "include",
  "int.mdx": "int",
  "math-courses.mdx": "math-courses",
  "no-doc.mdx": "no-doc",
  "one-of.mdx": "one-of",
  "phone-number.mdx": "phone-number",
  "public.mdx": "public",
  "record.mdx": "record",
  "roles.mdx": "roles",
  "science-courses.mdx": "science-courses",
  "search-role.mdx": "search-role",
  "semester.mdx": "semester",
  "skip.mdx": "skip",
  "specified-by.mdx": "specified-by",
  "string.mdx": "string",
  "time.mdx": "time",
  "un-weighted-gpa.mdx": "un-weighted-gpa",
  "unused-type.mdx": "unused-type",
  "update-gpa.mdx": "update-gpa",
  "update-grade.mdx": "update-grade",
  "url.mdx": "url",
  "weighted-gpa.mdx": "weighted-gpa",
}
`;

exports[`renderer generateDocFromSchema() Markdown document structure from GraphQL schema is correct when using flat hierarchy 2`] = `{}`;

exports[`renderer generateDocFromSchema() Markdown document structure from GraphQL schema is correct when using grouping 1`] = `
{
  "course/_category_.yml": "label: Course
position: 1
link: null
collapsible: true
collapsed: true
",
  "course/mutations/_category_.yml": "label: Mutations
position: 1
link: null
collapsible: true
collapsed: true
",
  "course/mutations/add-course.mdx": "add-course",
  "course/mutations/drop-course.mdx": "drop-course",
  "course/queries/_category_.yml": "label: Queries
position: 1
link: null
collapsible: true
collapsed: true
",
  "course/queries/all-courses.mdx": "all-courses",
  "course/queries/course.mdx": "course",
  "course/queries/math-courses.mdx": "math-courses",
  "course/queries/science-courses.mdx": "science-courses",
  "generated.md": "Dummy homepage for tweet.graphql",
  "grade/_category_.yml": "label: Grade
position: 1
link: null
collapsible: true
collapsed: true
",
  "grade/mutations/_category_.yml": "label: Mutations
position: 1
link: null
collapsible: true
collapsed: true
",
  "grade/mutations/update-gpa.mdx": "update-gpa",
  "grade/mutations/update-grade.mdx": "update-grade",
  "grade/queries/_category_.yml": "label: Queries
position: 1
link: null
collapsible: true
collapsed: true
",
  "grade/queries/gpa.mdx": "gpa",
  "grade/queries/un-weighted-gpa.mdx": "un-weighted-gpa",
  "grade/queries/weighted-gpa.mdx": "weighted-gpa",
  "misc/_category_.yml": "label: Misc
position: 1
link: null
collapsible: true
collapsed: true
",
  "misc/directives/_category_.yml": "label: Directives
position: 1
link: null
collapsible: true
collapsed: true
",
  "misc/directives/deprecated.mdx": "deprecated",
  "misc/directives/doc.mdx": "doc",
  "misc/directives/include.mdx": "include",
  "misc/directives/no-doc.mdx": "no-doc",
  "misc/directives/one-of.mdx": "one-of",
  "misc/directives/public.mdx": "public",
  "misc/directives/skip.mdx": "skip",
  "misc/directives/specified-by.mdx": "specified-by",
  "misc/enums/_category_.yml": "label: Enums
position: 1
link: null
collapsible: true
collapsed: true
",
  "misc/enums/roles.mdx": "roles",
  "misc/interfaces/_category_.yml": "label: Interfaces
position: 1
link: null
collapsible: true
collapsed: true
",
  "misc/interfaces/record.mdx": "record",
  "misc/objects/_category_.yml": "label: Objects
position: 1
link: null
collapsible: true
collapsed: true
",
  "misc/objects/course.mdx": "course",
  "misc/objects/department-information.mdx": "department-information",
  "misc/objects/semester.mdx": "semester",
  "misc/objects/unused-type.mdx": "unused-type",
  "misc/queries/_category_.yml": "label: Queries
position: 1
link: null
collapsible: true
collapsed: true
",
  "misc/queries/search-role.mdx": "search-role",
  "misc/scalars/_category_.yml": "label: Scalars
position: 1
link: null
collapsible: true
collapsed: true
",
  "misc/scalars/boolean.mdx": "boolean",
  "misc/scalars/date.mdx": "date",
  "misc/scalars/email-address.mdx": "email-address",
  "misc/scalars/id.mdx": "id",
  "misc/scalars/int.mdx": "int",
  "misc/scalars/phone-number.mdx": "phone-number",
  "misc/scalars/string.mdx": "string",
  "misc/scalars/time.mdx": "time",
  "misc/scalars/url.mdx": "url",
}
`;

exports[`renderer generateDocFromSchema() Markdown document structure from GraphQL schema is correct when using grouping 2`] = `{}`;

exports[`renderer generateDocFromSchema() generates Markdown document structure from GraphQL schema 1`] = `
{
  "directives/_category_.yml": "label: Directives
position: 1
link: null
collapsible: true
collapsed: true
",
  "directives/deprecated.mdx": "deprecated",
  "directives/include.mdx": "include",
  "directives/one-of.mdx": "one-of",
  "directives/skip.mdx": "skip",
  "directives/specified-by.mdx": "specified-by",
  "enums/_category_.yml": "label: Enums
position: 1
link: null
collapsible: true
collapsed: true
",
  "enums/media-type.mdx": "media-type",
  "generated.md": "Dummy homepage for tweet.graphql",
  "inputs/_category_.yml": "label: Inputs
position: 1
link: null
collapsible: true
collapsed: true
",
  "inputs/tweet-content.mdx": "tweet-content",
  "interfaces/_category_.yml": "label: Interfaces
position: 1
link: null
collapsible: true
collapsed: true
",
  "interfaces/node.mdx": "node",
  "mutations/_category_.yml": "label: Mutations
position: 1
link: null
collapsible: true
collapsed: true
",
  "mutations/create-tweet.mdx": "create-tweet",
  "mutations/delete-tweet.mdx": "delete-tweet",
  "mutations/mark-tweet-read.mdx": "mark-tweet-read",
  "objects/_category_.yml": "label: Objects
position: 1
link: null
collapsible: true
collapsed: true
",
  "objects/media.mdx": "media",
  "objects/meta.mdx": "meta",
  "objects/notification.mdx": "notification",
  "objects/stat.mdx": "stat",
  "objects/tweet.mdx": "tweet",
  "objects/user.mdx": "user",
  "queries/_category_.yml": "label: Queries
position: 1
link: null
collapsible: true
collapsed: true
",
  "queries/notifications-meta.mdx": "notifications-meta",
  "queries/tweet.mdx": "tweet",
  "queries/tweets-meta.mdx": "tweets-meta",
  "queries/tweets.mdx": "tweets",
  "queries/user.mdx": "user",
  "queries/users.mdx": "users",
  "scalars/_category_.yml": "label: Scalars
position: 1
link: null
collapsible: true
collapsed: true
",
  "scalars/boolean.mdx": "boolean",
  "scalars/date.mdx": "date",
  "scalars/id.mdx": "id",
  "scalars/int.mdx": "int",
  "scalars/sri.mdx": "sri",
  "scalars/string.mdx": "string",
  "scalars/url.mdx": "url",
  "subscriptions/_category_.yml": "label: Subscriptions
position: 1
link: null
collapsible: true
collapsed: true
",
  "subscriptions/notifications.mdx": "notifications",
}
`;
